### PR TITLE
Update to Async `BrowserContext`

### PR DESCRIPTION
**Pull request by inancgumus**
## What?

Update relevant `BrowserContext` methods from sync to async.

## Checklist

- [x] I have used a meaningful title for the PR.
- [x] I have described the changes I've made in the "What?" section above.
- [x] I have performed a self-review of my changes.
- [x] I have run the `npm start` command locally and verified that the changes look good.

- [x] I have made my changes in the `docs/sources/next` folder of the documentation.
- [ ] I have reflected my changes in the `docs/sources/v{most_recent_release}` folder of the documentation.
- [ ] I have reflected my changes in the relevant folders of the two previous k6 versions of the documentation (if still applicable to previous versions).

## Related PR(s)/Issue(s)

grafana/xk6-browser#1295

### CODE REVIEW

File: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/grantpermissions.md [hunk 1]
```diff
@@ -17,6 +17,12 @@ Grants specified permissions to the `BrowserContext`. Only grants corresponding
...
 
+### Returns
+
+| Type            | Description                                                     |
+| --------------- | --------------------------------------------------------------- |
+| `Promise<void>` | A Promise that fulfills when the permissions have been granted. |
+
 ### Example
 
 {{< code >}}
```
Comment (allansson) [inline docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/grantpermissions.md:28]: The examples seems to be missing an `await` in the call to `grantPermissions`.

File: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/cookies.md [hunk 2]
```diff
@@ -5,17 +5,17 @@ description: 'Retrieves context cookies.'
...
 
 ### Returns
 
-| Type  | Description                                                                                                                 |
-| ----- | --------------------------------------------------------------------------------------------------------------------------- |
-| array | A list of [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie). |
+| Type                     | Description                                                                                                                                                |
+| ------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `Promise<Array<Cookie>>` | A Promise that fulfills with an array of [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie). |
```
Comment (allansson) [inline docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/cookies.md:18]: It is common to use simple array types in typescript, i.e. `Cookie[]`, for simple types. The generic version is usually preserved for more complex types, e.g. `Array<typeof someVar>`, `Array<MyGeneric<string>>`.

TLDR; I think we should use `Promise<Inner[]>` as much as we can, because it's easier to read.

**Review by allansson — APPROVED** [review]
This is one heck of a job. 👍 

I found one example where we're missing an `await`, but other than that I couldn't find anything.

Left a suggestion on the return types as well.

File: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/cookies.md (hunk 2 above)
Comment (inancgumus) [reply docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/cookies.md:18]: Good point 👍 I follow PW's approach to be consistent with them. As people usually wonder how k6-browser is different compared to PW's. WDYT?

Comment (allansson) [reply docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/cookies.md:18]: > - How is k6-browser different from PW?
> - Our docs have better type signatures

Personally, I think the (arguably) improved readability outweighs being consistent with PW's docs. It also appears to be convention we've used in the type defs (although there aren't a lot of array types).

That said, it's not a hill I'm willing to die on.

Comment (inancgumus) [reply docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/cookies.md:18]: Created an issue to do this later #1593

File: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/grantpermissions.md (hunk 1 above)
Comment (inancgumus) [reply docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/grantpermissions.md:28]: Fixed here https://github.com/grafana/k6-docs/pull/1590/commits/0931b05878c947231f61e5449a6857496513f029

File: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/close.md [hunk 3]
```diff
@@ -29,9 +29,9 @@ export const options = {
 
 export default function () {
   const context = await browser.newContext();
-  context.newPage();
+await context.newPage();
```
Comment (ankur22) [inline docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/close.md]:
```suggestion
  await context.newPage();
```

File: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/pages.md [hunk 4]
```diff
@@ -42,10 +42,10 @@ export const options = {
 
 export default function () {
   const context = await browser.newContext();
-  context.newPage();
+await context.newPage();
```
Comment (ankur22) [inline docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/pages.md]:
```suggestion
  await context.newPage();
```

File: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/addcookies.md [hunk 5]
```diff
@@ -5,14 +5,20 @@ description: 'Clears context cookies.'
 
 # addCookies()
 
-Adds a list of [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie) into the [BrowserContext](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie). All pages within this [BrowserContext](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie) will have these [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie) set.
+Adds a list of [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie) into the [browser context](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/). All pages within this [browser context](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/) will have these [cookies](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/cookie) set.
```
Comment (ankur22) [inline docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/addcookies.md:8]: I'm not sure about the `BrowserContext` to `browser context` change. I think it makes it easier for the user to see it used in one way, so that they can easily understand the relationship between all the components.

Why did you make the change?

**Review by ankur22 — APPROVED** [review]
LGTM

File: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/addcookies.md (hunk 5 above)
Comment (inancgumus) [reply docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/addcookies.md:8]: It was easier to read in a sentence. We've been using this pattern in other places (before this PR), too. I've created an issue to keep everything consistent: https://github.com/grafana/k6-docs/issues/1596.

File: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/close.md (hunk 3 above)
Comment (inancgumus) [reply docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/close.md]: This is fixed somehow, IDK how :)
